validation/poststart_fail.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	tap "github.com/mndrix/tap-go"
+	rspec "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/opencontainers/runtime-tools/specerror"
+	"github.com/opencontainers/runtime-tools/validation/util"
+	uuid "github.com/satori/go.uuid"
+)
+
+func main() {
+	t := tap.New()
+	t.Header(0)
+
+	bundleDir, err := util.PrepareBundle()
+	if err != nil {
+		return
+	}
+	defer os.RemoveAll(bundleDir)
+
+	g := util.GetDefaultGenerator()
+	output := filepath.Join(bundleDir, g.Spec().Root.Path, "output")
+	poststart := rspec.Hook{
+		Path: filepath.Join(bundleDir, g.Spec().Root.Path, "/bin/false"),
+		Args: []string{"false"},
+	}
+	g.AddPostStartHook(poststart)
+	poststartOK := rspec.Hook{
+		Path: filepath.Join(bundleDir, g.Spec().Root.Path, "/bin/sh"),
+		Args: []string{
+			"sh", "-c", fmt.Sprintf("echo 'post-start called' >> %s", output),
+		},
+	}
+	g.AddPostStartHook(poststartOK)
+	poststop := rspec.Hook{
+		Path: filepath.Join(bundleDir, g.Spec().Root.Path, "/bin/sh"),
+		Args: []string{
+			"sh", "-c", fmt.Sprintf("echo 'post-stop called' >> %s", output),
+		},
+	}
```

Comment:
I don't think this `poststop` is needed. If the final output is not `post-start called\npost-stop called\n`, it's not immediately clear that it's poststart's error, and further judgment is needed.  So I think this `poststop` should be removed to remove unnecessary effects.

Reply:
you are right, updated.